cmd/snap-confine/snap-confine.c

```diff
@@ -63,6 +64,26 @@ void sc_maybe_fixup_permissions()
 	}
 }
 
+// sc_maybe_fixup_udev will remove incorrectly created udev tags
+// that cause libudev on 16.04 to fail with "udev_enumerate_scan failed".
+// See also:
+// https://forum.snapcraft.io/t/weird-udev-enumerate-error/2360/17
+void sc_maybe_fixup_udev()
+{
+	glob_t glob_res SC_CLEANUP(globfree) = {
+	.gl_pathv = NULL};
+	const char *glob_pattern = "/run/udev/tags/snap_*/*nvidia*";
+	int err = glob(glob_pattern, 0, NULL, &glob_res);
+	if (err != 0 && err != GLOB_NOMATCH) {
+		die("cannot search using glob pattern %s: %d",
+		    glob_pattern, err);
+	}
+	// killem'all
+	for (size_t i = 0; i < glob_res.gl_pathc; ++i) {
+		unlink(glob_res.gl_pathv[i]);
+	}
```

Comment:
While it would seem to make logical sense for glob() to set gl_pathc to 0 when GLOB_NOMATCH, I don't see in the man page that this is strictly guaranteed. I suspect your observed behavior is that it will be zero if not found. I'd feel slightly more comfortable and we'd get a teensy optimization if you only ran "killem'all" when err == GLOB_NOMATCH.